lib/init.coffee

```diff
@@ -102,23 +108,29 @@ module.exports =
       realpath = fs.realpathSync(p);
       return textEditor.getPath().substr(0, realpath.length) == realpath;
 
-  getFilesEndingWith: (startPath, endsWith) ->
+  getFilesEndingWith: (startPath, endsWith, ignoreFn) ->
     foundFiles = []
-    if !fs.existsSync(startPath)
-      return foundFiles
-    files = fs.readdirSync(startPath)
-    i = 0
-    while i < files.length
-      filename = path.join(startPath, files[i])
-      stat = fs.lstatSync(filename)
-      if stat.isDirectory()
-        foundFiles.push.apply(foundFiles, @getFilesEndingWith(filename, endsWith))
-      else if filename.indexOf(endsWith, filename.length - (endsWith.length)) >= 0
-        foundFiles.push.apply(foundFiles, [filename])
-        #Array::push.apply foundFiles, filename
-      i++
-    return foundFiles
-
+    folderFiles = []
+    voucher fs.readdir, startPath
+      .then (files) =>
+        folderFiles = files
+        Promise.all files.map (f) =>
```

Comment:
Although that does make it closer to ES6, it's not the best practice for CoffeeScript which is what this file is written in. It does make a difference in the [generated JS](http://coffeescript.org/#try:Promise.all%20files.map%20(f)%20-%3E%0A%20%20%20%20%20%20%20%20%20%20filename%20%3D%20path.join%20startPath%2C%20f%0A%20%20%20%20%20%20%20%20%20%20voucher%20fs.lstat%2C%20filename%0A%0APromise.all%20files.map%20(f)%20%3D%3E%0A%20%20%20%20%20%20%20%20%20%20filename%20%3D%20path.join%20startPath%2C%20f%0A%20%20%20%20%20%20%20%20%20%20voucher%20fs.lstat%2C%20filename) code as well.

Also `coffeelint` complains about it :stuck_out_tongue:.